docs/reference/api/c-api.md

```diff
@@ -66,20 +66,32 @@ is as follows
-      * See test ```TestSharedAllocatorUsingCreateAndRegisterAllocator``` in
-     https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/shared_lib/test_inference.cc for an example.
-      * Configuring *OrtArenaCfg*:
-         * Default values for these configs can be found in the [BFCArena class](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/bfc_arena.h).
-         * ```initial_chunk_size_bytes```: This is the size of the region that the arena allocates first. Chunks are handed over to allocation requests from this region. If the logs show that the arena is getting extended a lot more than expected, you're better off choosing a big enough initial size for this.
+      * See test ```TestSharedAllocatorUsingCreateAndRegisterAllocator``` in https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/shared_lib/test_inference.cc for an example.
+      * Configuring *OrtArenaCfg* (use API ```CreateArenaCfgV2``` to create ```OrtArenaCfg``` instance from ORT release 1.8 onwards, prior releases used the now-deprecated ```CreateArenaCfg``` to create the instance):
+         * ```CreateArenaCfgV2``` takes the following a list of keys as described below and takes a corresponding set of values (one for each key). Default values for these configs can be found in the [BFCArena class](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/bfc_arena.h). See test ```ConfigureCudaArenaAndDemonstrateMemoryArenaShrinkage``` in https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/shared_lib/test_inference.cc for an example usage of ```CreateArenaCfgV2```.
          * ```max_mem```: This is the maximum amount of memory the arena allocates. If a chunk cannot be serviced by any existing region, the arena extends itself by allocating one more region depending on available memory (max_mem - allocated_so_far). An error is returned if available memory is less than the requested extension.
          * ```arena_extend_strategy```: This can take only 2 values currently: kSameAsRequested or kNextPowerOfTwo. As the name suggests kNextPowerOfTwo (the default) extends the arena by a power of 2, while kSameAsRequested extends by a size that is the same as the allocation request each time. kSameAsRequested is suited for more advanced configurations where you know the expected memory usage in advance.
+         * ```initial_chunk_size_bytes```: This config is only relevant when the arena extend strategy is kNextPowerOfTwo. This is the (possible) size of the region that the arena allocates in the very first allocation (if the first memory request is greater than this the allocation size will be larger than this provided value). Chunks are handed over to allocation requests from this region. If the logs show that the arena is getting extended a lot more than expected, you're better off choosing a big enough initial size for this.
```

Comment:
>This config is only relevant when the arena extend strategy is kNextPowerOfTwo [](http://example.com/codeflow?start=43&length=78)

Side question: Is there a task in the backlog to simplify the usage of the initial chunk size (regardless of growth strategy, if the initial allocation can be satisfied by initial_chunk_size_bytes, we use that value).

Reply:
Added: https://msdata.visualstudio.com/Vienna/_workitems/edit/1209620

2 proposals for enhancement of the existing arena class:

1) initial chunk to be honored by both extension strategies (and their behaviors diverge only on "extension" allocations)

2) For 'kNextPowerOfTwo', the second allocation stops using 2*initial chunk as a candidate because it could be a very high value depending on what the user sets the initial chunk to be. So the proposal is to use initial_growth_chunk_size_bytes to extend on.